pkg/ee/kubevirt-network-controller/controller.go

```diff
@@ -0,0 +1,203 @@
+      code of the software licensed under this license, and, to the extent
+      explicitly provided under this license, the binary code.
+   2.	Any use of the software which exceeds the foregoing right, including,
+      without limitation, its execution, compilation, copying, modification
+      and distribution, is expressly prohibited.
+   3.	THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND,
+      EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+      MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+      IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY
+      CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+      TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+      SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+   END OF TERMS AND CONDITIONS
+*/
+
+package kubevirtnetworkcontroller
+
+import (
+	"context"
+	"fmt"
+
+	kubeovnv1 "github.com/kubeovn/kube-ovn/pkg/apis/kubeovn/v1"
+	"go.uber.org/zap"
+
+	kubermaticv1 "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1"
+	kubermaticv1helper "k8c.io/kubermatic/v2/pkg/apis/kubermatic/v1/helper"
+	predicateutil "k8c.io/kubermatic/v2/pkg/controller/util/predicate"
+	"k8c.io/kubermatic/v2/pkg/provider"
+	"k8c.io/kubermatic/v2/pkg/version/kubermatic"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/record"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	ctrlruntimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	ControllerName                = "kubevirt-network-controller"
+	WorkloadSubnetLabel           = "k8c.io/kubevirt-workload-subnet"
+	NetworkPolicyPodSelectorLabel = "cluster.x-k8s.io/cluster-name"
+)
+
+type Reconciler struct {
+	ctrlruntimeclient.Client
+
+	seedGetter   provider.SeedGetter
+	configGetter provider.KubermaticConfigurationGetter
+
+	log        *zap.SugaredLogger
+	versions   kubermatic.Versions
+	workerName string
+	recorder   record.EventRecorder
+}
+
+func Add(
+	mgr manager.Manager,
+	log *zap.SugaredLogger,
+
+	numWorkers int,
+	workerName string,
+
+	seedGetter provider.SeedGetter,
+	configGetter provider.KubermaticConfigurationGetter,
+
+	versions kubermatic.Versions,
+) error {
+	reconciler := &Reconciler{
+		log:          log.Named(ControllerName),
+		Client:       mgr.GetClient(),
+		seedGetter:   seedGetter,
+		configGetter: configGetter,
+		workerName:   workerName,
+		recorder:     mgr.GetEventRecorderFor(ControllerName),
+	}
+
+	_, err := builder.ControllerManagedBy(mgr).
+		Named(ControllerName).
+		WithOptions(controller.Options{
+			MaxConcurrentReconciles: numWorkers,
+		}).
+		For(&kubermaticv1.Cluster{}, builder.WithPredicates(predicateutil.Factory(func(o ctrlruntimeclient.Object) bool {
```

Comment:
Don't you need also to watch for subnets?

Reply:
As discussed early, we will drop this idea for now, as we will keep the subnets in the seed provider network settings